norddrop/src/device/mod.rs

```diff
@@ -203,6 +203,89 @@ impl NordDropFFI {
         })
     }
 
+    pub(super) fn purge_transfers(&mut self, transfer_ids: &str) -> Result<()> {
+        trace!(
+            self.logger,
+            "norddrop_purge_transfers() : {:?}",
+            transfer_ids
+        );
+
+        let transfer_ids: Vec<String> =
+            serde_json::from_str(transfer_ids).map_err(|_| ffi::types::NORDDROP_RES_JSON_PARSE)?;
+
+        self.rt.block_on(async {
+            self.instance
+                .lock()
+                .await
+                .as_mut()
+                .ok_or(ffi::types::NORDDROP_RES_NOT_STARTED)?
+                .purge_transfers(transfer_ids)
+                .await
+                .map_err(|err| {
+                    error!(self.logger, "Failed to purge transfers: {:?}", err);
+                    match err {
+                        drop_transfer::Error::StorageError => ffi::types::NORDDROP_RES_DB_ERROR,
+                        _ => ffi::types::NORDDROP_RES_DB_ERROR,
+                    }
+                })
+        })
+    }
+
+    pub(super) fn purge_transfers_until(&mut self, until_timestamp: i64) -> Result<()> {
+        trace!(
+            self.logger,
+            "norddrop_purge_transfers_until() : {:?}",
+            until_timestamp
+        );
+
+        self.rt.block_on(async {
+            self.instance
+                .lock()
+                .await
+                .as_mut()
+                .ok_or(ffi::types::NORDDROP_RES_NOT_STARTED)?
+                .purge_transfers_until(until_timestamp)
+                .await
+                .map_err(|err| {
+                    error!(self.logger, "Failed to purge transfers: {:?}", err);
+                    match err {
+                        drop_transfer::Error::StorageError => ffi::types::NORDDROP_RES_DB_ERROR,
+                        _ => ffi::types::NORDDROP_RES_DB_ERROR,
+                    }
+                })
+        })
+    }
+
+    pub(super) fn get_transfers_since(&mut self, since_timestamp: i64) -> Result<String> {
+        trace!(
+            self.logger,
+            "norddrop_get_transfers_since() since_timestamp: {:?}",
+            since_timestamp
+        );
+
+        let result = self.rt.block_on(async {
+            self.instance
+                .lock()
+                .await
+                .as_mut()
+                .ok_or(ffi::types::NORDDROP_RES_NOT_STARTED)?
+                .get_transfers_since(since_timestamp)
+                .await
+                .map_err(|err| {
+                    error!(self.logger, "Failed to get transfers: {:?}", err);
+                    match err {
+                        drop_transfer::Error::StorageError => ffi::types::NORDDROP_RES_DB_ERROR,
+                        _ => ffi::types::NORDDROP_RES_DB_ERROR,
+                    }
+                })
+        })?;
+
+        let result =
+            serde_json::to_string(&result).map_err(|_| ffi::types::NORDDROP_RES_JSON_PARSE)?;
+
+        Ok(result)
+    }
+
```

Comment:
These are wrappers for wrappers. I propose keeping the `Storage` field in the `instance` and calling it directly. It will be less code and less mental overhead IMHO.

Reply:
You mean adding a reference to `StorageDispatch` in `NorddropFFI` to skip going trough `Service` for the calls?